### PR TITLE
update spf13/cobra dependency to 1.0.0

### LINKED ISSUE #91021: Bump spf13/cobra dependency to 1.0.0 version
Cobra 1.0.0 release was released a month ago:
https://github.com/spf13/cobra/releases/tag/v1.0.0

We should bump our vendor to use that (instead of 0.5 version we currently use).

As a bonus - in 0.6 version, support for "Context" was added, which would help me with cleaning up UnitWithoutRetry - see https://github.com/kubernetes/kubernetes/pull/90815

/help wanted
/kind cleanup

### PR DESCRIPTION
**Pull request by gongguan**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes [#91021](https://github.com/kubernetes/kubernetes/issues/91021)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wojtek-t** [comment]
Thanks a lot @gongguan - this lgtm.

Though I don't feel competent to review go modules changes - will wait for @liggitt to take a look.

File: LICENSES/vendor/github.com/russross/blackfriday/v2/LICENSE [hunk 1]
```diff
@@ -0,0 +1,33 @@
+= vendor/github.com/russross/blackfriday/v2 licensed under: =
+
+Blackfriday is distributed under the Simplified BSD License:
```
Comment (liggitt) [inline LICENSES/vendor/github.com/russross/blackfriday/v2/LICENSE:3]: for reference, this is the 2-clause BSD license (https://github.com/cncf/foundation/blob/master/whitelist-policy.md#approved-licenses-for-whitelist)

File: go.mod [hunk 2]
```diff
@@ -201,13 +201,14 @@ replace (
...
 	github.com/containernetworking/cni => github.com/containernetworking/cni v0.7.1
 	github.com/coredns/corefile-migration => github.com/coredns/corefile-migration v1.0.6
+	github.com/coreos/bbolt => github.com/coreos/bbolt v1.3.2
 	github.com/coreos/etcd => github.com/coreos/etcd v3.3.10+incompatible
-	github.com/coreos/go-etcd => github.com/coreos/go-etcd v2.0.0+incompatible
 	github.com/coreos/go-oidc => github.com/coreos/go-oidc v2.1.0+incompatible
 	github.com/coreos/go-semver => github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd => github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
-	github.com/coreos/pkg => github.com/coreos/pkg v0.0.0-20180108230652-97fdf19511ea // 97fdf19511ea is the SHA for git tag v4
+	github.com/coreos/pkg => github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f // 97fdf19511ea is the SHA for git tag v4
```
Comment (liggitt) [inline go.mod:209]: comment no longer matches the pinned sha... update comment to the new tag 399ea9e2e55f corresponds to (if any) or drop the comment if there is no tag

File: go.mod [hunk 3]
```diff
@@ -365,27 +366,29 @@ replace (
 	github.com/rogpeppe/fastuuid => github.com/rogpeppe/fastuuid v0.0.0-20150106093220-6724a57986af
 	github.com/rubiojr/go-vhd => github.com/rubiojr/go-vhd v0.0.0-20160810183302-0bfd3b39853c
 	github.com/russross/blackfriday => github.com/russross/blackfriday v1.5.2
+	github.com/russross/blackfriday/v2 => github.com/russross/blackfriday/v2 v2.0.1
 	github.com/satori/go.uuid => github.com/satori/go.uuid v1.2.0
 	github.com/seccomp/libseccomp-golang => github.com/seccomp/libseccomp-golang v0.9.1
 	github.com/sergi/go-diff => github.com/sergi/go-diff v1.0.0
+	github.com/shurcooL/sanitized_anchor_name => github.com/shurcooL/sanitized_anchor_name v1.0.0
```
Comment (liggitt) [inline go.mod:373]: this was dropped in 1.16... it's unfortunate this got reintroduced

File: vendor/github.com/spf13/cobra/command.go [hunk 4]
```diff
@@ -994,7 +1030,11 @@ func (c *Command) InitDefaultVersionFlag() {
 		} else {
 			usage += c.Name()
 		}
-		c.Flags().Bool("version", false, usage)
+		if c.Flags().ShorthandLookup("v") == nil {
+			c.Flags().BoolP("version", "v", false, usage)
```
Comment (liggitt) [inline vendor/github.com/spf13/cobra/command.go:1034]: does this cause any issues with our usage of `-v` to control log verbosity?

**Review by liggitt — COMMENTED** [review]
this touches bash autocompletion... please ensure kubectl autocompletion still works properly

File: LICENSES/vendor/github.com/russross/blackfriday/v2/LICENSE (hunk 1 above)
Comment (gongguan) [reply LICENSES/vendor/github.com/russross/blackfriday/v2/LICENSE:3]: Sorry for not good at dependency area, I'll take a deeper look at it and try to reply your comments tomorrow(my timezone).

Comment (liggitt) [reply LICENSES/vendor/github.com/russross/blackfriday/v2/LICENSE:3]: no change was required here, I was just noting the license was acceptable

File: go.mod (hunk 2 above)
Comment (gongguan) [reply go.mod]: dropped the comment because I haven't found 399ea9e2e55f in latest tag(v4)

File: vendor/github.com/spf13/cobra/command.go (hunk 4 above)
Comment (gongguan) [reply vendor/github.com/spf13/cobra/command.go:1034]: test with `-v` locally, no issue happened.

**Review by soltysh — APPROVED** [review]
Thanks for doing this!

**Pull request merged**